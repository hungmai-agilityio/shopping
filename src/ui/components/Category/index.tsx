'use client';

import { useState } from 'react';
import { clsx } from 'clsx';
import { useRouter, useSearchParams } from 'next/navigation';

// Constants
import { mada } from '@/constants';

interface CategoryProps {
  categories: string[];
  queryParam: string;
}

export const Category = ({ categories, queryParam }: CategoryProps) => {
  const { push } = useRouter();
  const searchParams = useSearchParams();
  const params = new URLSearchParams(searchParams.toString());
  const initCategory = searchParams.get(queryParam) || '';

  const [currentCategory, setCurrentCategory] = useState<string>(initCategory);

  // Update category and URL when a new category is selected
  const handleCategorySelect = (category: string) => {
    params.set(queryParam, category);
    push(`?${params.toString()}`, { scroll: false });
    setCurrentCategory(category);
  };

  return (
    <>
      <ul
        className={`${mada.className} my-5 flex xl:gap-10 gap-5 justify-center text-dark`}
      >
        {categories.map((category, index) => (
          <li
            className={clsx(
              'relative capitalize cursor-pointer hover:text-second',
              {
                'text-second font-bold': currentCategory === category
              }
            )}
            key={index}
            onClick={handleCategorySelect.bind(null, category)}
          >
            {category}
            {currentCategory === category && (
              <span className="absolute left-0 right-0 -bottom-2 mx-auto h-line w-full bg-active"></span>
            )}
          </li>
        ))}
      </ul>
    </>
  );
};
