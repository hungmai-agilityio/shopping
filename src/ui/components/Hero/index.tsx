import { ReactNode } from 'react';
import { clsx } from 'clsx';

// Components
import Breadcrumb, { BreadcrumbItem } from '@/ui/components/common/BreadCrumb';

interface HeroProps {
  isHeroSecond?: boolean;
  isBreadCrumb?: boolean;
  styles?: string;
  children: ReactNode;
  items?: BreadcrumbItem[];
}

const Hero = ({
  isHeroSecond = false,
  isBreadCrumb,
  styles,
  children,
  items
}: HeroProps) => {
  return (
    <div
      className={clsx(
        'relative w-full h-full overflow-hidden py-10 px-28',
        styles
      )}
    >
      {isBreadCrumb && (
        <Breadcrumb
          items={items!}
          color="text-white"
          styles="relative container z-20 md:flex hidden"
        />
      )}
      <div className="relative z-10">{children}</div>

      <div
        className={clsx('absolute inset-0 bg-gradient-primary ', {
          'top-0 left-0 right-0 bottom-0 -skew-y-6 origin-top-left max-h-hero':
            isHeroSecond
        })}
      ></div>
    </div>
  );
};

export default Hero;
