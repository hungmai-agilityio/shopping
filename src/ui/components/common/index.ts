export { default as BreadCrumb } from '@/ui/components/common/BreadCrumb';
export { default as Button } from '@/ui/components/common/Button';
export { default as Badge } from '@/ui/components/common/Badge';
export { default as RadioButton } from '@/ui/components/common/RadioButton';
export { default as Icon } from '@/ui/components/common/Icon';
export { default as Input } from '@/ui/components/common/Input';
export { default as InputController } from '@/ui/components/common/Input/InputController';
export { default as Logo } from '@/ui/components/common/Logo';
export { default as Table } from '@/ui/components/common/Table';
export { default as Typography } from '@/ui/components/common/Typography';
export { default as Quantity } from '@/ui/components/common/Quantity';
export { default as ColorPicker } from '@/ui/components/common/ColorPicker';
export { default as ToastMessage } from '@/ui/components/common/ToastMessage';
export { default as Spinner } from '@/ui/components/common/Loading/Spinner'
export { default as Brand } from '@/ui/components/common/Brand'
export { default as Modal } from '@/ui/components/common/Modal';
export { default as ModalDelete } from '@/ui/components/common/Modal/Delete';
export { default as ModalAuth } from '@/ui/components/common/Modal/Auth';
export { default as Dropdown } from '@/ui/components/common/Dropdown';
export { default as DropdownNote } from '@/ui/components/common/Dropdown/Note';
export { default as CardProductSkeleton } from '@/ui/components/common/Skeleton/CardProduct';
export { default as Carousel } from '@/ui/components/common/Carousel';
export { default as CarouselCategories } from '@/ui/components/common/Carousel/Categories';
export { default as CarouselProduct } from '@/ui/components/common/Carousel/Product';
export { default as BurgerMenu } from '@/ui/components/common/BurgerMenu';
export { default as CardImage } from '@/ui/components/common/Card/Image';
export { default as CardProduct } from '@/ui/components/common/Card/Product';
export { default as CardProfile } from '@/ui/components/common/Card/Profile';
export { default as Card } from '@/ui/components/common/Card';
export { default as List } from '@/ui/components/common/List';
export { default as Navbar } from '@/ui/components/common/Navbar';

export * from '@/ui/components/common/Table/Body';
export * from '@/ui/components/common/Table/Row';
export * from '@/ui/components/common/Table/Row/Empty';
export * from '@/ui/components/common/Table/Cell';
export * from '@/ui/components/common/Table/Head';
export * from '@/ui/components/common/Typography/Heading';
