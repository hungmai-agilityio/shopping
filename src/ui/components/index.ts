export { default as UserAction } from '@/ui/components/UserAction';
export { default as UserIcon } from '@/ui/components/UserAction/Icon';
export { default as CartProduct } from '@/ui/components/Cart/Product';
export { default as CartItem } from '@/ui/components/Cart/Product/Item';
export { default as CartCheckout } from '@/ui/components/Cart/Checkout';
export { default as AboutInfo } from '@/ui/components/AboutInfo';
export { default as AuthForm } from '@/ui/components/Auth';
export { default as WishList } from '@/ui/components/WishList';
export { default as Hero } from '@/ui/components/Hero';
export { default as Tabs } from '@/ui/components/common/Tabs';

export * from '@/ui/components/common';
export * from '@/ui/components/Category';
export * from '@/ui/components/common/Tabs/Panel';
