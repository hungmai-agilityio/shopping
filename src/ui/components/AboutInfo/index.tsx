import { FONT_SIZE, FONT_WEIGHT, mada, TAG } from '@/constants';
import CardImage from '@/ui/components/Card/Image';
import { Typography } from '@/ui/components/common';
import { clsx } from 'clsx';

interface AboutInfoProps {
  desc: string;
  isReverse?: boolean;
  aboutName: string;
  src: string;
  title: string;
  styles?: string;
}

const AboutInfo = ({
  title,
  desc,
  aboutName,
  src,
  isReverse,
  styles
}: AboutInfoProps) => {
  return (
    <div
      className={clsx(
        'lg:flex block justify-between w-full gap-10',
        {
          'flex-row-reverse': isReverse
        },
        styles
      )}
    >
      <div className="lg:max-w-card-sm lg:text-left text-center">
        <Typography
          fontWeight={FONT_WEIGHT.BOLD}
          tag={TAG.H2}
          size={FONT_SIZE.MEDIUM}
          className={`${mada.className} capitalize my-5`}
        >
          {title}
        </Typography>
        <Typography className="my-6 text-dark-md" size={FONT_SIZE.X_SMALL}>
          {desc}
        </Typography>
      </div>
      <div className="flex justify-center">
        <CardImage
          src={src}
          alt={aboutName}
          width={590}
          height={349}
          title={aboutName}
          isTitleLeft={isReverse}
          className="w-full h-96"
        />
      </div>
    </div>
  );
};

export default AboutInfo;
