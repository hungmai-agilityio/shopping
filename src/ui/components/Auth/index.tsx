import { ReactNode } from 'react';
import { BreadCrumb, Typography } from '@/ui/components/common';
import { FONT_SIZE, FONT_WEIGHT, TAG, mada, popping } from '@/constants';

interface AuthFormProps {
  title: string;
  subtitle: ReactNode;
  backgroundImage: string;
  children: ReactNode;
}

const AuthForm = ({
  title,
  subtitle,
  backgroundImage,
  children
}: AuthFormProps) => {
  const breadCrumb = [
    { label: 'Home', href: '/' },
    { label: 'My Account', href: '' }
  ];

  return (
    <div
      className={`${popping.className} bg-no-repeat bg-cover text-white`}
      style={{ backgroundImage: `url(${backgroundImage})` }}
    >
      <div className="bg-gradient-overlay py-20">
        <div className="w-full md:max-w-card-sm md:mx-40 mx-5">
          <BreadCrumb items={breadCrumb} color="text-white" />
          <Typography
            color="text-white"
            fontWeight={FONT_WEIGHT.BOLD}
            tag={TAG.H1}
            size={FONT_SIZE.MEDIUM}
            className={`${mada.className} mt-8 text-5xl`}
          >
            My Account
          </Typography>
          <Typography
            color="text-white"
            fontWeight={FONT_WEIGHT.BOLD}
            tag={TAG.H2}
            size={FONT_SIZE.MEDIUM}
            className={`${mada.className} mt-11`}
          >
            {title}
          </Typography>

          <div className="mt-6">{subtitle}</div>
          <div className="mt-10 mb-20">{children}</div>
        </div>
      </div>
    </div>
  );
};

export default AuthForm;
