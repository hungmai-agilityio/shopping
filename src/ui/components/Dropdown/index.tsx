'use client';

import { ReactNode, useRef, useState } from 'react';
import { clsx } from 'clsx';

import { Icon } from '@/ui/components/common';
import { useOutsideClick } from '@/hooks';

interface DropdownProps {
  children: ReactNode;
  styles?: string;
  setIsNoteVisible?: (visible: boolean) => void;
  isNoteVisible?: boolean;
}
const Dropdown = ({
  children,
  styles,
  setIsNoteVisible,
  isNoteVisible
}: DropdownProps) => {
  const dropdownRef = useRef<HTMLDivElement>(null);

  const toggleNoteVisibility = () =>
    setIsNoteVisible && setIsNoteVisible(!isNoteVisible);

  useOutsideClick({
    ref: dropdownRef,
    handler: () => setIsNoteVisible && setIsNoteVisible(false)
  });

  return (
    <div ref={dropdownRef}>
      <div
        className={clsx('flex gap-5 items-center', styles)}
        onClick={toggleNoteVisibility}
      >
        <p className="text-second text-lg font-bold cursor-pointer lowercase">
          products notes
        </p>
        <Icon src="/arrow.svg" alt="arrow icon" width={24} height={24} />
      </div>
      {isNoteVisible && children}
    </div>
  );
};

export default Dropdown;
