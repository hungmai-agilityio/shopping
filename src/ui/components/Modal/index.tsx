import { Button } from '@/ui/components/common';
import { ReactNode } from 'react';

interface ModalProps {
  isOpen: boolean;
  onClose: () => void;
  title?: string;
  children: ReactNode;
  isConfirm?: boolean;
  buttonName?: string;
  onConfirm?: () => void;
}

const Modal = ({
  isOpen,
  onClose,
  title,
  children,
  isConfirm,
  buttonName,
  onConfirm
}: ModalProps) => {
  if (!isOpen) return null;

  return (
    <div className="fixed inset-0 flex items-center justify-center z-50">
      <div
        className="fixed inset-0 bg-black opacity-50"
        onClick={onClose}
      ></div>
      <div className="bg-white rounded-lg shadow-lg max-w-md w-full z-10">
        <div className="p-4 border-b">
          <h2 className="text-lg font-bold">{title}</h2>
        </div>
        <div className="p-4">{children}</div>
        <div className="p-4 border-t flex justify-between">
          <Button
            className="bg-red-500 text-white py-2 px-4 rounded"
            onClick={onClose}
          >
            Close
          </Button>
          {isConfirm && <Button onClick={onConfirm}>{buttonName}</Button>}
        </div>
      </div>
    </div>
  );
};

export default Modal;
