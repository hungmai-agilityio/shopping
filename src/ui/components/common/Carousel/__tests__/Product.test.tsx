import { render, screen, fireEvent } from '@testing-library/react';
import { mockProducts } from '@/mocks';
import { CarouselProduct } from '@/ui/components';

describe('CarouselProduct', () => {
  const onOrderMock = jest.fn();

  beforeEach(() => {
    render(<CarouselProduct products={mockProducts} onOrder={onOrderMock} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders with default props', () => {
    expect(screen.getByText('Basic T-Shirt')).toBeInTheDocument();
    expect(
      screen.getByText('A comfortable and stylish basic t-shirt.')
    ).toBeInTheDocument();
  });

  test('displays the next product when the next button is clicked', () => {
    fireEvent.click(screen.getByAltText('icon arrow right'));

    expect(screen.getByText('Sporty T-Shirt')).toBeInTheDocument();
    expect(
      screen.getByText('Perfect for your workout sessions.')
    ).toBeInTheDocument();
  });

  test('displays the previous product when the previous button is clicked', () => {
    fireEvent.click(screen.getByAltText('icon arrow left'));

    expect(screen.getByText('Graphic T-Shirt')).toBeInTheDocument();
    expect(
      screen.getByText('Unique graphic design for a standout look.')
    ).toBeInTheDocument();
  });

  test('calls onOrder function when "Order Now" button is clicked', () => {
    const orderButton = screen.getByText('Order Now');
    fireEvent.click(orderButton);
    expect(onOrderMock).toHaveBeenCalledTimes(1);
  });

  it('should be render match to snapshot', () => {
    const { container } = render(
      <CarouselProduct products={mockProducts} onOrder={onOrderMock} />
    );

    expect(container).toMatchSnapshot();
  });
});
