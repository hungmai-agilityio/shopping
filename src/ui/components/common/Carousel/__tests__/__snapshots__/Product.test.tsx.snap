// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselProduct should be render match to snapshot 1`] = `
<div>
  <div
    class="className w-full h-full bg-gradient-primary lg:flex block justify-between py-20 xl:pr-32 pr-5"
  >
    <div
      class="flex"
    >
      <div
        class="w-full flex items-center justify-center"
      >
        <div
          class="transition-all duration-300 ease-in-out mb-2 relative"
        >
          <div
            class="relative inline-flex items-center justify-center"
          >
            <img
              alt="Sporty T-Shirt"
              class="cursor-pointer w-52 h-32"
              data-nimg="1"
              decoding="async"
              height="130"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FWtgcmWR%2Ft-shirt-2.webp&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FWtgcmWR%2Ft-shirt-2.webp&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FWtgcmWR%2Ft-shirt-2.webp&w=640&q=75 2x"
              style="color: transparent; object-fit: cover; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="200"
            />
          </div>
          <img
            alt="icon Ellipse"
            class="absolute -left-1 mt-5"
            data-nimg="1"
            decoding="async"
            height="21"
            loading="lazy"
            src="/ellipse.svg"
            style="color: transparent;"
            width="321"
          />
        </div>
        <div
          class="relative w-card-sm flex items-center justify-center"
        >
          <div
            class="absolute left-10"
          >
            <div
              class="relative inline-flex items-center justify-center cursor-pointer"
            >
              <img
                alt="icon arrow left"
                class="bg-white p-1 z-10"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="40"
                src="/arrow-left.svg"
                style="color: transparent;"
                width="40"
              />
            </div>
          </div>
          <div
            class="mt-5"
          >
            <div
              class="relative inline-flex items-center justify-center"
            >
              <img
                alt="Basic T-Shirt"
                class="transition-transform duration-300 ease-in-out w-slide-sm h-slide-img-md"
                data-nimg="1"
                decoding="async"
                height="428"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FnjkCQHq%2Ft-shirt-3.webp&w=640&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FnjkCQHq%2Ft-shirt-3.webp&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FnjkCQHq%2Ft-shirt-3.webp&w=640&q=75 2x"
                style="color: transparent; object-fit: cover; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                width="314"
              />
            </div>
            <img
              alt="icon Ellipse"
              class="mx-auto mt-5"
              data-nimg="1"
              decoding="async"
              height="21"
              loading="lazy"
              src="/ellipse.svg"
              style="color: transparent;"
              width="314"
            />
          </div>
          <div
            class="absolute right-10"
          >
            <div
              class="relative inline-flex items-center justify-center cursor-pointer"
            >
              <img
                alt="icon arrow right"
                class="bg-white p-1 z-10"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="40"
                src="/arrow-right.svg"
                style="color: transparent;"
                width="40"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex-1 ml-10 flex flex-col justify-center max-w-card-sm"
    >
      <h2
        class="font-bold text-white md:text-2xl text-md className xl:text-5xl text-4xl"
      >
        Basic T-Shirt
      </h2>
      <div
        class="flex justify-between mb-6"
      >
        <div>
          <p
            class="font-bold text-white lg:text-lg text-sm"
          >
            Color
          </p>
          <div
            class="mt-4 flex items-center gap-3"
          >
            <button
              aria-label="White-picker"
              class="h-4 w-4 rounded-full focus:border focus:border-green-800 border border-dark"
              id="White"
              style="background-color: White;"
            />
            <button
              aria-label="Blue-picker"
              class="h-4 w-4 rounded-full focus:border focus:border-green-800"
              id="Blue"
              style="background-color: Blue;"
            />
            <button
              aria-label="Green-picker"
              class="h-4 w-4 rounded-full focus:border focus:border-green-800"
              id="Green"
              style="background-color: Green;"
            />
          </div>
        </div>
        <div
          class="ml-4 flex items-center"
        >
          <div
            class="relative inline-flex items-center justify-center"
          >
            <img
              alt="rating"
              data-nimg="1"
              decoding="async"
              height="24"
              loading="lazy"
              src="star.svg"
              style="color: transparent;"
              width="152"
            />
          </div>
          <span
            class="font-thin text-white text-sm"
          >
            ( 
            10k
             )
          </span>
        </div>
      </div>
      <div
        class="flex justify-between items-center mb-4"
      >
        <div
          class="mb-6"
        >
          <span
            class="font-bold text-white lg:text-lg text-sm"
          >
            Size
          </span>
          <div
            class="flex gap-4 mt-2"
          >
            <button
              class="w-6 h-6 bg-white rounded-full drop-shadow-lg text-sm focus:text-second focus:font-bold"
            >
              S
            </button>
            <button
              class="w-6 h-6 bg-white rounded-full drop-shadow-lg text-sm focus:text-second focus:font-bold"
            >
              M
            </button>
            <button
              class="w-6 h-6 bg-white rounded-full drop-shadow-lg text-sm focus:text-second focus:font-bold"
            >
              L
            </button>
          </div>
        </div>
        <span
          class="font-semibold text-white md:text-2xl text-md className xl:text-heading text-4xl"
        >
          15
          $
        </span>
      </div>
      <p
        class="font-thin text-gray-200 text-sm my-5"
      >
        A comfortable and stylish basic t-shirt.
      </p>
      <button
        class="truncate md:h-12 h-8 md:text-lg md:w-btn-md w-28 text-xs bg-white text-purple-500 font-bold"
      >
        Order Now
      </button>
    </div>
  </div>
</div>
`;
